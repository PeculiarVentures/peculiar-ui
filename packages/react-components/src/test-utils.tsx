import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
