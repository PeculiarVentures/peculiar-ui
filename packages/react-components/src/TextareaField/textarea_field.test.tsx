import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { TextareaField } from './index';

describe('<TextareaField />', () => {
  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(<TextareaField id="test-id" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have label', () => {
    const {
      asFragment,
    } = render(
      <TextareaField
        label="Test label"
        id="test-id"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be disabled', () => {
    const {
      asFragment,
    } = render(<TextareaField disabled id="test-id" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have required', () => {
    const {
      asFragment,
    } = render(<TextareaField required id="test-id" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have name attr', () => {
    const {
      asFragment,
    } = render(<TextareaField name="test-name" id="test-id" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have className', () => {
    const {
      asFragment,
    } = render(
      <TextareaField
        className="div-cls"
        id="test-id"
        inputProps={{
          className: 'input-cls',
        }}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
