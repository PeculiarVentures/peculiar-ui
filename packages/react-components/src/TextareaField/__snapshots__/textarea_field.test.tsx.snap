// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextareaField /> should be disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: var(--pv-size-base) var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  min-height: var(--pv-size-base-14);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <textarea
      class="emotion-0"
      disabled=""
      id="test-id"
    />
  </div>
</DocumentFragment>
`;

exports[`<TextareaField /> should have className 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: var(--pv-size-base) var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  min-height: var(--pv-size-base-14);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div
    class="div-cls"
  >
    <textarea
      class="input-cls emotion-0"
      id="test-id"
    />
  </div>
</DocumentFragment>
`;

exports[`<TextareaField /> should have label 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-bottom: 2px;
  display: inline-block;
}

.emotion-1 {
  margin: 0;
  color: var(--pv-color-gray-10);
  font-weight: var(--pv-text-c2-weight);
  font-size: var(--pv-text-c2-size);
  line-height: var(--pv-text-c2-height);
  letter-spacing: var(--pv-text-c2-spacing);
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: var(--pv-size-base) var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  min-height: var(--pv-size-base-14);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-2::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-2:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-2:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-2:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <label
      class="emotion-0"
      for="test-id"
      id="test-id-label"
    >
      <span
        class="emotion-1"
      >
        Test label
      </span>
    </label>
    <textarea
      class="emotion-2"
      id="test-id"
    />
  </div>
</DocumentFragment>
`;

exports[`<TextareaField /> should have name attr 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: var(--pv-size-base) var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  min-height: var(--pv-size-base-14);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <textarea
      class="emotion-0"
      id="test-id"
      name="test-name"
    />
  </div>
</DocumentFragment>
`;

exports[`<TextareaField /> should have required 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: var(--pv-size-base) var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  min-height: var(--pv-size-base-14);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <textarea
      class="emotion-0"
      id="test-id"
      required=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextareaField /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: var(--pv-size-base) var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  min-height: var(--pv-size-base-14);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <textarea
      class="emotion-0"
      id="test-id"
    />
  </div>
</DocumentFragment>
`;
